src/accelerate/utils/modeling.py

```diff
@@ -322,15 +322,22 @@ def get_max_memory(max_memory: Optional[Dict[Union[int, str], Union[int, str]]]
     Get the maximum memory available if nothing is passed, converts string to int otherwise.
     """
     import psutil
-
+     
     if max_memory is None:
         if not torch.cuda.is_available():
             max_memory = {}
+        if not is_xpu_available():
+            max_memory = {}
```

Comment:
This will replace `max_memory` on any setup having GPUs but not XPUs, so we can't change the file like this.

Reply:
Resolved for now.